capsicum-test-main.cc

```diff
@@ -6,10 +6,46 @@
 #include <ctype.h>
 #include <errno.h>
 #include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
 #include <iostream>
 #include "gtest/gtest.h"
 #include "capsicum-test.h"
 
+std::string tmpdir;
+
+class SetupEnvironment : public ::testing::Environment
+{
+public:
+  SetupEnvironment() : teardown_tmpdir_(false) {}
+  void SetUp() override {
+    if (tmpdir.empty()) {
+      std::cerr << "Generating temporary directory root: ";
+      CreateTemporaryRoot();
+    } else {
+      std::cerr << "User provided temporary directory root: ";
+    }
+    std::cerr << tmpdir << std::endl;
+  }
+  void CreateTemporaryRoot() {
+    char *tmp_root;
+    char *tmp_root_template = tempnam(nullptr, "capsicum_test.XXXXXX");
```

Comment:
This hits a Linux/FreeBSD difference.  FreeBSD [has](https://www.freebsd.org/cgi/man.cgi?tempnam(3)):
> The argument prefix, if non-NULL, is used to specify a file name prefix, which will be the first part of the created file name.

but Linux [has](http://man7.org/linux/man-pages/man3/tempnam.3.html):

> The filename suffix of the pathname generated will start with pfx in case pfx is a non-NULL string of at most five bytes.

In practice on Linux this gives a name like `/tmp/capsiXLaAS8` and then the `mkdtemp()` call below fails (no XXXXXX suffix any more).

Reply:
You're right -- I was doing the wrong thing.

I've shortened the prefix slightly to ensure the path is intuitive, but stays within the limits of the API.